game/src/main/org/apollo/game/model/Position.java

```diff
@@ -224,6 +224,66 @@ public boolean isWithinDistance(Position other, int distance) {
 		return deltaX <= distance && deltaY <= distance && getHeight() == other.getHeight();
 	}
 
+	/**
+	 * Gets a position, based upon this position, from the given deltas.
+	 *
+	 * @param deltaX The delta x.
+	 * @param deltaY The delta y.
+	 * @return The position.
+	 */
+	public Position fromDeltas(int deltaX, int deltaY) {
+		return fromDeltas(deltaX, deltaY, 0);
+	}
+
+	/**
+	 * Gets a position, based upon this position, from the given deltas.
+	 *
+	 * @param deltaX The delta x.
+	 * @param deltaY The delta y.
+	 * @param deltaZ The delta z.
+	 * @return The new position.
+	 */
+	public Position fromDeltas(int deltaX, int deltaY, int deltaZ) {
+		return new Position(getX() + deltaX, getY() + deltaY, getHeight() + deltaZ);
+	}
+
+	/**
+	 * Gets the adjacent position in the given direction.
+	 *
+	 * @param direction the direction
+	 * @return the position
+	 */
+	public Position fromDirection(Direction direction) {
```

Comment:
```
public Position add(Direction direction) {
```
Also use the helper functions in Direction: https://github.com/apollo-rsps/apollo/blob/bugfix/collision-detection/game/src/main/org/apollo/game/model/Direction.java#L190